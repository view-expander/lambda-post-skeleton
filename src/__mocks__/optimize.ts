export const DUMMY_OPTIMIZED_IMAGE = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256"><path fill="#ccc" fill-rule="evenodd" d="M0 66.096v66.096l57.25-.305c63.71-.339 61.754-.53 65.442 6.38 1.587 2.973 1.701 7.828 1.421 60.483l-.305 57.25h131.947l.729-125.25c.401-68.888.524-126.6.273-128.25-.252-1.65-.525 25.013-.607 59.25L256 124l-46.25.25c-65.578.354-67.778.304-71.531-1.637-6.828-3.531-6.652-1.717-6.33-65.363l.29-57.25H0v66.096m.472.404c0 36.3.125 51.005.278 32.677.152-18.327.152-48.027-.001-66C.597 15.205.472 30.2.472 66.5"/></svg>`

export default jest.fn().mockResolvedValue(DUMMY_OPTIMIZED_IMAGE)
