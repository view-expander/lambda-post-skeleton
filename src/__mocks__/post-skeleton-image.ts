export const DUMMY_POST_RESPONSE = {
  $response: {
    httpResponse: {
      statusCode: 200,
    },
  },
}
export default jest.fn().mockResolvedValue(DUMMY_POST_RESPONSE)
