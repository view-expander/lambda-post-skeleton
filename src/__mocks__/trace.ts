export default jest
  .fn()
  .mockResolvedValue(
    '<svg width="8" height="8" fill="none" xmlns="http://www.w3.org/2000/svg"><g clip-path="url(#clip0)"><path fill="#fff" d="M0 0h8v8H0z"/><path fill="#000" d="M0 0h4v4H0zM4 4h4v4H4z"/></g><defs><clipPath id="clip0"><path fill="#fff" d="M0 0h8v8H0z"/></clipPath></defs></svg>'
  )
