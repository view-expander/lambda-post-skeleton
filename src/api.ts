import axios from 'axios'

const api = axios.create({
  baseURL: process.env.API_PATH,
  timeout: 180000,
})

export default api
