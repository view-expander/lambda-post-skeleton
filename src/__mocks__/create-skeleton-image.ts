import { DUMMY_OPTIMIZED_IMAGE } from './optimize'
export default jest.fn().mockResolvedValue(DUMMY_OPTIMIZED_IMAGE)
