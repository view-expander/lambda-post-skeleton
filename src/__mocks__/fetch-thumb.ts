import MockAdapter from 'axios-mock-adapter'
import api from '../api'
export const DUMMY_IMAGE = new Uint8Array(
  Buffer.from(
    '/9j/4AAQSkZJRgABAgEASABIAAD/2wBDAAEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/2wBDAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/wAARCAAIAAgDAREAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwCP/lS6/wCsk3/Dyb/uzr/hTH/DHX/iU/8AwsX/AIWL/wANT/8AUi/8Ij/wgv8AzM//AAk//FPAAP/Z',
    'base64'
  ).buffer
).buffer

export default jest.fn().mockImplementation(async (key: string) => {
  const mock = new MockAdapter(api)
  mock
    .onGet(`/photo/${key}`)
    .reply(200, DUMMY_IMAGE, { 'Content-Type': 'image/jpeg' })

  const res = await api.get(`/photo/${key}`, {
    params: {
      size: 'thumb',
    },
    responseType: 'arraybuffer',
  })

  mock.restore()

  return Promise.resolve(res)
})
