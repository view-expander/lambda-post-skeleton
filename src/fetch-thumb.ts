import api from './api'

export default (key: string) =>
  api.get<ArrayBuffer>(`/photo/${key}`, {
    params: {
      size: 'thumb',
    },
    responseType: 'arraybuffer',
  })
