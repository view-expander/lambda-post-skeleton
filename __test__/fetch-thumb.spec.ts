import { DUMMY_IMAGE } from '../src/__mocks__/fetch-thumb'
import MockAdapter from 'axios-mock-adapter'
import api from '../src/api'
import fetchThumb from '../src/fetch-thumb'

describe('fetch-thumb()', () => {
  let mock: MockAdapter

  beforeEach(() => {
    mock = new MockAdapter(api)
  })

  afterEach(() => {
    mock.restore()
  })

  it('resolves arrayBuffer', async () => {
    expect.assertions(1)

    mock.onGet('/source/image.jpg').reply(200, DUMMY_IMAGE, {
      'content-type': 'image/jpeg',
    })

    return expect(fetchThumb('image.jpg')).resolves.toMatchSnapshot()
  })
})
