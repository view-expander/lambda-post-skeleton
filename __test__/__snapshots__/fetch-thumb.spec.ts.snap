// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch-thumb() resolves arrayBuffer 1`] = `
Object {
  "config": Object {
    "baseURL": "https://api.view-expander.net/v1",
    "data": undefined,
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxContentLength": -1,
    "method": "get",
    "params": Object {
      "size": "thumb",
    },
    "responseType": "arraybuffer",
    "timeout": 180000,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "url": "/photo/image.jpg",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": ArrayBuffer [],
  "headers": Object {
    "Content-Type": "image/jpeg",
  },
  "request": Object {
    "responseUrl": "/photo/image.jpg",
  },
  "status": 200,
}
`;
